### PR TITLE
Recipe App - Back end

### PR DESCRIPTION
**Pull request by Sanja969**
### Recipe App - Back end

We followed the layout of the wireframes provided.
image

- Login page and registration page:
          - Built with Devise.
- Food list:
          - Display a list of food added by the logged-in user as in the wireframe
          - Lead to a form that allows users to add new food.
- Recipes list:
           - Display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
            - If the user is the owner of the recipe, the user can delete it.
- Public recipe list:
             - Should display a list of all public recipes ordered by newest as in the wireframe.
             - Lead to recipe details.
- Recipe details:
             - If the recipe is public or the user is the owner of the recipe, display the recipe details as in the wireframe.
             - If the user is the owner of the recipe, it leads to the form that allows the user to add new food.
- General shopping list view
              - Show the list of food that is missing for all recipes of the logged-in user (compared the list of food for all recipes with 
the general food list of that user).
               - Count the total food items and total price of the missing food.
- Made sure there are no N+1 queries happening.
- Created a navigation menu that allows users to open all of the pages we created.

### CODE REVIEW

File: test/models/food_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (rustam-tolipov) [inline test/models/food_test.rb:1]: - [x] You made a great effort for making this project. However, there is a requirement for students must write `unite tests` and `integration tests` for the project. [Highlight about test](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/buisness_requirements.md#:~:text=Set%20up%20RSpec%20and%20Capybara%20testing%20libraries.)

Screen Shot 2022-10-14 at 20 00 48

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+header {
+  background-color: black;
+  height: 70px;
+  width: 100%;
+  margin-bottom: 100px;
+  color: white;
+  // display: flex;
+  // align-items: center;
+  // justify-content: space-between;
```
Comment (rustam-tolipov) [inline app/assets/stylesheets/application.scss]: - [x] Kindly remove commented codes from your file.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Great Design_


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @Sanja969 and @NabilHY,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**